lgc/state/PalMetadata.cpp

```diff
@@ -47,6 +47,16 @@
 using namespace lgc;
 using namespace llvm;
 
+namespace {
+template <class T> msgpack::ArrayDocNode BuildArrayDocNode(msgpack::Document *document, llvm::ArrayRef<T> data) {
```

Comment:
I think we normally have camelBack not CamelCase for even a global function.

Reply:
Done and I added comments to the function.